Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -1114,33 +1248,234 @@ def export_vulnerabilities_command(args: Dict[str, Any]) -> PollResult:
+        headers=list(context_to_hr),
+        headerTransform=context_to_hr.get,
+        removeNull=True)
+
+
+def list_scan_filters_command(client: Client) -> CommandResults:
+
+    response_dict = client.list_scan_filters()
+    filters = response_dict.get('filters', [])
+
+    return CommandResults(
+        outputs_prefix='TenableIO.ScanFilter',
+        outputs_key_field='name',
+        outputs=filters,
+        readable_output=scan_filters_human_readable(filters),
+        raw_response=response_dict)
+
+
+def scan_history_readable(history: list) -> str:
+    context_to_hr = {
+        'id': 'History id',
+        'scan_uuid': 'History uuid',
+        'status': 'Status',
+        'is_archived': 'Is archived',
+        'custom': 'Targets custom',
+        'default': 'Targets default',
+        'visibility': 'Visibility',
+        'time_start': 'Time start',
+        'time_end': 'Time end',
+    }
+    return tableToMarkdown(
+        'Tenable IO Scan History',
+        [d | d.get('targets', {}) for d in history],
+        headers=list(context_to_hr),
+        headerTransform=context_to_hr.get,
+        removeNull=True)
+
+
+def scan_history_params(args: dict) -> dict:
+    return {
+        'sort': ','.join(
+            f'{field}:{order}'
+            for field, order
+            in zip(
+                argToList(args.get('sortFields')),
+                # * 3 makes it apply to all sortFields even if it has only one value
+                argToList(args.get('sortOrder')) * 3)),
+        'exclude_rollover': args['excludeRollover'],
+    } | sub_dict(args, 'limit', 'offset')
+
+
+@paginate(
+    page_size_arg_name='pageSize',
+    keys_to_pages='history')
+def scan_history_request(args: dict[str, Any], client: Client):
+    return client.get_scan_history(
+        args['scanId'],
+        scan_history_params(args))
+
+
+def get_scan_history_command(*args) -> CommandResults:
+
+    history = scan_history_request(*args)
+
+    return CommandResults(
+        outputs_prefix='TenableIO.ScanHistory',
+        outputs_key_field='id',
+        outputs=history,
+        readable_output=scan_history_readable(history))
+
+
+def build_filters(args: dict) -> dict:
+    '''
+    Makes filter args for the export scan endpoint by sequentially matching
+    the filterName, filterQuality and filterValue lists in args.
+    Example:
+
+        filterName:     name,description
+        filterQuality:  =,!=
+        filterValue:    test,test2
+
+        returns:
+        {
+            'filter.0.filter': 'name',
+            'filter.0.quality': '=',
+            'filter.0.value': 'test',
+            'filter.1.filter': 'description',
+            'filter.1.quality': '!=',
+            'filter.1.value': 'test2',
+        }
+    '''
+    filters: dict[str, str] = {}
+    filter_lists = map(argToList, map(args.get, ('filterName', 'filterQuality', 'filterValue')))
+    for i, (name, quality, value) in enumerate(zip(*filter_lists)):
```

Comment:
lets treat each one individually, then we wont need to enumerate like this

Reply:
ok